tutorial/write-custom-widget.md

```diff
@@ -107,3 +107,178 @@ Take care not to keep any important state in your renderer - animation tickers
 are well suited to that location but user state would not be. A widget that is
 hidden may have it's renderer destroyed and if it is shown again the new renderer
 must be able to reflect the same widget state.
+
+### Example Widget
+
+The following covers in detail the creation of a simple Widget and it's Renderer.
+
+#### Create a Widget
+
+So the widget requires the`widget.BaseWidget`. For example:
+
+```go
+type MyWidget struct {
+	widget.BaseWidget           // Inherit from BaseWidget
+	minSize           fyne.Size // The minimum size of the widget
+	text              string	// The text to display in the widget
+}
+```
+
+To construct the widget create a New<widgetname> function. The example we are using is called MyWidget so by convention the following constructor function name would be used:
+
+```go
+func NewMyWidget(W, H float32, text string) *MyWidget {
+	w := &MyWidget{ 							 // Initialise the widget storage
+		text:    text,                           // Save the text
+		minSize: fyne.Size{Width: W, Height: H}, // Set the size
+	}
+	w.ExtendBaseWidget(w) // Required to initialise the base widget
+	return w
+}
+```
+
+We need to override the MinSize() function so we can return the minimum size of our widget. 
+
+```go
+func (w *MyWidget) MinSize() fyne.Size {
+	return w.minSize
+}
+func (w *MyWidget) Resize(s fyne.Size) { // See note below
+	w.BaseWidget.Resize(s)
+}
+```
+
+Note : It is not necessary to override the Resize function (shown above) but if you need to change the Resize behaviour of your widget and you add a ```Resize(s fyne.Size)``` function then you **must** include the ```w.BaseWidget.Resize(s)```. in the code.
+
+The final function to add is the ```CreateRenderer() fyne.WidgetRenderer``` function:
+
+```go
+func (w *MyWidget) CreateRenderer() fyne.WidgetRenderer {
+	return newMyWidgetRenderer(w.minSize, w.text)
```

Comment:
This is not best form - you are capturing the state of the widget when creating the renderer - this means that changes in state will never be reflected in the UI.
Instead just pass the widget in, so the renderer can access whichever state is needed.

Reply:
You have fixed the `newMyWidgetRenderer` code below, but this function call is still passing the old parameters